src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+    F::AbstractAlgebra.Ring = ZZ, saturate::Bool=false, simplify::Bool=true,
+    char::Union{Int,Nothing}=nothing, q::Union{Int,Nothing}=nothing)::MatroidRealizationSpace
+
+    if char!=nothing && !isprime(char) && char!=0
+        error("The characteristic has to be 0 or a prime number.")
+    end
+
+    #Construct the base ring as F_p if q=p^k
+    if q!=nothing
+        isprimepower, p, k = is_prime_power_with_data(q)
+        if !isprimepower
+            error("The given q has to be a prime power.")
+        end
+        if char!=nothing && char!=p
+            error("The given characteristic doesn't match q.")
+        else
+            char = p
+        end
+    end
+
+    if char == 0
+        F = QQ
+    elseif char != nothing
+        F = GF(char)
+    end
+
+    rk = rank(M)
+    n = length(M)
+
+    goodM = isomorphic_matroid(M, [i for i in 1:n])
+
+    Bs = bases(goodM)
+    
+    if !isnothing(B) 
+        goodB = sort!(Int.([M.gs2num[j] for j in B]))
+    else
+        goodB = find_good_basis_heuristically(goodM)
+    end
+    polyR, mat = realization_space_matrix(goodM, goodB, F)
+
+    
+    eqs = Vector{RingElem}()
+    ineqs = Vector{RingElem}()
+
+    #need to catch the corner-case if there are no variables at all
+    if !(typeof(polyR) <: MPolyRing)
+        RS = MatroidRealizationSpace(ideal(polyR,0), ineqs, polyR, mat, F, char, q)
+        RS.realizable = true
+        return RS
+    end
+    
+    for col in subsets(Vector(1:n),rk)
+        
+        col_det = det(mat[:,col])
+        
+        if total_degree( col_det ) <= 0 
+            
+            if col_det != 0 && col in Bs 
+                if isunit( col_det ) 
+                    continue
+                end
+            elseif col_det != 0 # and col is not a basis
+                # determinant nonzero but set not a basis
+                push!( eqs, col_det )
+            elseif col in Bs 
+                #determinant zero but set is a basis, i.e. M is not realizable
+                return MatroidRealizationSpace(F, char, q)
+            else
+                continue
+            end
+            
+        end
+        
+        if  col in Bs
+            push!( ineqs, col_det )
+        else
+            push!( eqs, col_det )
+        end
+        
+    end
+
+    if q != nothing
+        for i in 1:polyR.nvars
+            push!(eqs, polyR[i]^q-polyR[i])
+        end
+    end
+
+    def_ideal = ideal(polyR,eqs)
+    def_ideal = ideal(groebner_basis(def_ideal))
+    isone(def_ideal) && return MatroidRealizationSpace(F, char, q)
+
+    # Unclear if we should use this. Can be too slow in some cases.
+    #=
+    if !iszero(def_ideal)
+        for i in 1:length(ineqs)
+            ineqs[i] = Oscar.reduce(ineqs[i], gens(def_ideal))
+        end
+    end
+    (polyR(0) in ineqs) && return MatroidRealizationSpace(nothing, nothing, nothing, nothing, false, F, char, q)
+    =#
```

Comment:
```suggestion
```

Reply:
looks like this was deleted, as requested.